ClientSyncComponent/ClientSyncComponent.Client/PuzzleEntry.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Azure;
+using Azure.Data.Tables;
+
+namespace ClientSyncComponent.Client
+{
+    /// <summary>
+    /// Azure Table entity for puzzle state
+    /// </summary>
+    public class PuzzleEntry : ITableEntity
+    {
+        // Default constructor required by Azure query API
+        public PuzzleEntry()
+        {
+        }
+        
+        // todo morganb: translate from string player name to playerId
+        public PuzzleEntry(int puzzleId, int teamId, int playerId, string subPuzzleId, string locationKey, string propertyKey, string value, string channel)
+        {
+            PartitionKey = CreatePartitionKey(puzzleId, teamId);
+            RowKey = CreateRowKey(subPuzzleId, locationKey, propertyKey);
+            SubPuzzleId = subPuzzleId;
+            LocationKey = locationKey;
+            PropertyKey = propertyKey;
+            Value = value;
+            PlayerId = playerId;
+            Channel = channel;
+        }
+
+        private static string CreateRowKey(string subPuzzleId, string locationKey, string propertyKey)
```

Comment:
Which one of these two functions should `channel` go into? If it should be part of the PartitionKey, add placeholder space for it now?

Reply:
Will put in a todo for when we implement channels